pkg/solana/logpoller/loader.go

```diff
@@ -451,12 +448,9 @@ func (p *orderedParser) sendReadySlots() error {
 			continue
```

Comment:
I see... you'll probably want to delay the `heap.Pop` call I have above till here, so that if you break or return early it's not already removed. Instead of the initial `heap.Pop` you can "peak" at the minimum block number on the heap with `block := (*p.blocks)[0]`. Even though the rest may be out of order, this should always be guaranteed to be the min element.

I think the two places you'll need the `heap.Pop` are in place of the 2 instances of `rmvIdx = append(rmvIndex, idx)`, does that sound right?

Reply:
I'm not sure the `heap.Pop` call is useful in this instance though. Pop will remove the first element, but all I want to do is iterate through them in order without alterations. Then, when I want to remove specific indexes, Pop will not work. I need to use `heap.Remove` to specify which index to remove.

Reply:
Maybe I'm missing something, but is there a reason why you can't just Pop each block as soon as you've verified that the expectations are met? I don't see any case where you go back and look at old blocks or look ahead at future blocks, so I'm not sure why we have to flag the block for removal and then go back through later and locate all the ones flagged for removal? If we're done with it, why not just discard it and move on?